src/Domain/Block.cpp

```diff
@@ -98,8 +125,21 @@ std::ostream& operator<<(std::ostream& os,
 template <size_t VolumeDim>
 bool operator==(const Block<VolumeDim>& lhs,
                 const Block<VolumeDim>& rhs) noexcept {
+  // Since the external boundary conditions are all abstract base classes we
+  // can't compare them. We at least ensure that they match in terms of the
+  // number of external boundaries set.
   return lhs.id() == rhs.id() and lhs.neighbors() == rhs.neighbors() and
          lhs.external_boundaries() == rhs.external_boundaries() and
+         alg::all_of(rhs.external_boundaries(),
```

Comment:
I don't see how this tests anything about the equivalence of Blocks since `lhs.external_boundaries() == rhs.external_boundaries()`.  It will only fail for an invalid Block which should be caught at construction...

Reply:
heh, yea I think you're right. The other test I can add is that the `typeid` of the dereferenced base class pointers are the same to make sure that at the least the boundary conditions are of the same type. Thoughts?

Reply:
seems reasonable